presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/PlanPrinter.java

```diff
@@ -1406,21 +1400,22 @@ private static String formatHash(Optional<Symbol>... hashes)
         return "[" + Joiner.on(", ").join(symbols) + "]";
     }
 
-    private static String formatFrame(WindowFrame frame)
+    private static String formatFrame(WindowNode.Frame frame)
     {
         StringBuilder builder = new StringBuilder(frame.getType().toString());
-        FrameBound start = frame.getStart();
-        if (start.getValue().isPresent()) {
-            builder.append(" ").append(start.getOriginalValue().get());
+
+        Optional<Symbol> startValue = frame.getStartValue();
+        if (startValue.isPresent()) {
+            builder.append(" ").append(frame.getOriginalStartValue().get());
         }
-        builder.append(" ").append(start.getType());
+        builder.append(" ").append(frame.getStartType());
 
-        Optional<FrameBound> end = frame.getEnd();
-        if (end.isPresent()) {
-            if (end.get().getOriginalValue().isPresent()) {
-                builder.append(" ").append(end.get().getOriginalValue().get());
+        Optional<Symbol> endValue = frame.getEndValue();
+        if (endValue.isPresent()) {
+            if (frame.getOriginalEndValue().isPresent()) {
+                builder.append(" ").append(frame.getOriginalEndValue().get());
             }
-            builder.append(" ").append(end.get().getType());
+            builder.append(" ").append(frame.getEndType());
```

Comment:
`frame.getEndType()` is always present. Shouldn't it be printed unconditionally, just like `frame.getStartType()` is?